RentallCarsAPI/Controllers/CarController.cs

```diff
@@ -18,86 +16,175 @@ public class CarController : Controller
+                    car.Mark = model.Mark;
+                    car.Model = model.Model;
+                    car.Doors = model.Doors;
+                    car.Color = model.Color;
+                    response.Data = car;
+                    break;
                 }
-                
+            }
+                       
+
+            try
+            {
+                var writer = JsonConvert.SerializeObject(cars, Formatting.Indented);
+                System.IO.File.WriteAllText("cars.txt", writer);
+                response.Succes = true;
             }
             catch (Exception ex)
             {
-                oResponse.Message = "Serch error";
+                response.Message = "Impossible to Update";
+                return BadRequest(response);
             }
-
-            return Ok(oResponse);
+            response.Message = "Successfully updated";
+            return Ok(response);
         }
 
+        [HttpDelete("{id}")]
+        public IActionResult Delete(int id)
+        {
+            var response = new Response();
 
+            var cars = GetAll();
+            if (cars == null)
+            {
+                response.Message = "File reading failed";
+                return BadRequest(response);
+            }
+            foreach (var aux in cars)
+            {
+                if (aux.Id == id)
+                {
+                    cars.Remove(aux);
+                    try
+                    {
+                        var writer = JsonConvert.SerializeObject(cars, Formatting.Indented);
+                        System.IO.File.WriteAllText("cars.txt", writer);
+                        response.Succes = true;
+                    }
+                    catch (Exception ex)
+                    {
+                        response.Message = "Impossible to Delete";
+                        return BadRequest(response);
+                    }
+                    response.Succes = true;
+                    response.Data = aux;
+                    response.Message = "Delete successfully";
+                    return Ok(response);
+                }
+            }
+            response.Message = $"Car with id: {id} not found";
+            return NotFound(response);
+        }
+
+        private bool ValidateParams(CarRequest model)
+        {
+            bool flag = false;
+            if ((int)model.Mark >= 0 && (int)model.Mark <= 5)
+            {
+                if((int)model.Transmition >= 0 && (int)model.Transmition <= 1)
+                {
+                    flag = true;
+                }                    
+            }    
+            return flag;
+        }
         private List<Car> GetAll() {
             var cars = new List<Car>();
             try
             {
                 string list = System.IO.File.ReadAllText("cars.txt");
-                cars = JsonConvert.DeserializeObject<List<Car>>(list);                
+                cars = JsonConvert.DeserializeObject<List<Car>>(list);
```

Comment:
Podemos aprovechar este PR para retornar de forma directa el resultado de la deserialización y eliminar la variable `cars`:
```suggestion
                return JsonConvert.DeserializeObject<List<Car>>(list);
```
Qué te parece?